### PR TITLE
wqueue: Fix build error in HP/LP worker thread's

### PR DESCRIPTION
**Pull request by satheesh-ks**
Build failed for "stm32" boards with below error 

configs:
stm32f429i-disco/hello
stm32f407-disc1/hello

CC:  work_queue.c
work_queue.c: In function 'work_qqueue':
work_queue.c:137:2: error: 'next_work' undeclared (first use in this function)
  next_work = NULL;
  ^~~~~~~~~
CC: work_process.c
work_process.c: In function 'work_process':
work_process.c:277:28: error: 'set' undeclared (first use in this function)
   DEBUGVERIFY(sigwaitinfo(&set, NULL));

CC: work_process.c: In function 'work_process':
work_process.c:257:4: error: 'remaining' undeclared (first use in this function)
    remaining = work->delay - elapsed;
    ^~~~~~~~~

Signed-off-by: Satheesh Kumar <satheesh.ks@samsung.com>

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [cfc67249ced199d57e9045ab33b1a812c9f3f08a] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [cfc67249ced199d57e9045ab33b1a812c9f3f08a] - Code Rule Check (C++) OK.

**Comment by jeongchanKim** [comment]
I agree this changes. But this commit is not only for stm32f407. Please change commit message(title) for general.

File: os/wqueue/work_queue.c [hunk 1]
```diff
@@ -134,7 +134,9 @@ int work_qqueue(FAR struct wqueue_s *wqueue, FAR struct work_s *work, worker_t w
 #endif
 
 	/* check whether requested work is in queue list or not */
+#ifdef CONFIG_SCHED_WORKQUEUE_SORTING
 	next_work = NULL;
```
Comment (sijupunnnoose) [inline os/wqueue/work_queue.c]: How about initializing at line no 125 instead?

**Comment by seinfra** [comment]
Target : [220170cf21c88693c2df78da3592d0148aabf6ad] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [220170cf21c88693c2df78da3592d0148aabf6ad] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [ab9313586fa6b0dfa2b4ebf95c934ce9b742dfe2] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [ab9313586fa6b0dfa2b4ebf95c934ce9b742dfe2] - Code Rule Check (C++) OK.

File: os/wqueue/work_process.c [hunk 2]
```diff
@@ -274,7 +274,9 @@ void work_process(FAR struct wqueue_s *wqueue, int wndx)
 #endif
 		/* Wait indefinitely until signalled with SIGWORK */
 		wqueue->worker[wndx].busy = false;
+#ifdef CONFIG_SCHED_WORKQUEUE_SORTING
```
Comment (sunghan-chang) [inline os/wqueue/work_process.c]: This looks strange. This functionality is not related with sorting. This is for waiting new work. Could you check this conditional?

Comment (sunghan-chang) [reply os/wqueue/work_process.c]: I re-check the commit which merged kernel and user workqueue. This is wrong. Could you make a change?

Comment (satheesh-ks) [reply os/wqueue/work_process.c]: But the declaration of this variable(set) was under that specific flag.
Thats the reason, we made the changes accordingly. Pls refer below code

void work_process(FAR struct wqueue_s *wqueue, int wndx)
{
	volatile FAR struct work_s *work;
	worker_t worker;
	FAR void *arg;
	clock_t elapsed;
	clock_t stick;
	clock_t ctick;
	clock_t next;

**#ifdef CONFIG_SCHED_WORKQUEUE_SORTING**
	sigset_t set;
	**sigemptyset(&set);**
	sigaddset(&set, SIGWORK);
#else 
	clock_t remaining;
#endif

Comment (sunghan-chang) [reply os/wqueue/work_process.c]: @satheesh-ks I mean that code is wrong. I know why you add that conditional. But because original is wrong, yours is also wrong. That's why I suggest modify together.

Comment (sunghan-chang) [reply os/wqueue/work_process.c]: @satheesh-ks I sent you messages for that through the chat. Could you check that?

**Comment by seinfra** [comment]
Target : [1520e36b86b3e1e22c7b870ed6460fca5ddb760a] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [1520e36b86b3e1e22c7b870ed6460fca5ddb760a] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [563804451d6b7585d7e6dbcfc3222b9c9f4a5d80] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [563804451d6b7585d7e6dbcfc3222b9c9f4a5d80] - Code Rule Check OK.

File: os/wqueue/work_process.c [hunk 3]
```diff
@@ -134,11 +134,12 @@ void work_process(FAR struct wqueue_s *wqueue, int wndx)
 	clock_t ctick;
 	clock_t next;
 
-#ifdef CONFIG_SCHED_WORKQUEUE_SORTING
+#if (defined(CONFIG_SCHED_LPWORK) && CONFIG_SCHED_LPNTHREADS > 0) || defined(CONFIG_SCHED_WORKQUEUE_SORTING)
 	sigset_t set;
 	sigemptyset(&set);
 	sigaddset(&set, SIGWORK);
```
Comment (sunghan-chang) [inline os/wqueue/work_process.c]: Regardless of this PR, is this necessary all the time? Isn't it enough setting once at start of work queue?
